src/swsssdk/configdb.py

```diff
@@ -198,14 +200,44 @@ def mod_entry(self, table, key, data):
         if data == None:
             client.delete(_hash)
         else:
-            client.hmset(_hash, self.__typed_to_raw(data))
+            # We allow deletion of field by setting them to None, __typed_to_raw()
+            # will fill keys to be deleted in dkeys arg.
+            dkeys = dict()
+            raw = self.__typed_to_raw(data, dkeys)
+            if raw and len(raw):
```

Comment:
shouldn't "if raw" enough?

Reply:
_typed_to_raw may return raw as {}, if user opts to delete fields without adding any new field. Empty raw is not accepted by hmset.